packages/util-use-case/lib/enhancers/UseCaseWithLogger.test.mjs

```diff
@@ -1,8 +1,30 @@
-import test from '@zorko-io/tool-test-harness'
 import sinon from 'sinon'
+import {AssertionError} from 'assert'
+import test from '@zorko-io/tool-test-harness'
+import {CoreLogger} from '@zorko-io/util-logger'
 import {UseCaseWithLogger} from './UseCaseWithLogger'
 import {UseCase} from '../core'
-import {CoreLogger} from '@zorko-io/util-logger'
+
+test('create UseCase custom logger without origin', async (t) => {
+  const log = sinon.createStubInstance(CoreLogger)
+  const origin = sinon.createStubInstance(UseCase)
+  const name = 'TestUseCase'
+
+  await (async () => new UseCaseWithLogger({log, name}))().catch((err) => {
```

Comment:
Did you try [throwAsync](https://github.com/avajs/ava/blob/master/docs/03-assertions.md#throwsasyncthrower-expectation-message) ?

Reply:
done